arch/arm/core/aarch32/cortex_m/vector_table.S

```diff
@@ -38,37 +38,45 @@ SECTION_SUBSEC_FUNC(exc_vector_table,_vector_table_section,_vector_table)
 
     .word z_arm_hard_fault
 #if defined(CONFIG_ARMV6_M_ARMV8_M_BASELINE)
-    .word z_arm_reserved
-    .word z_arm_reserved
-    .word z_arm_reserved
-    .word z_arm_reserved
-    .word z_arm_reserved
-    .word z_arm_reserved
-    .word z_arm_reserved
+    .word 0
+    .word 0
+    .word 0
+    .word 0
+    .word 0
+    .word 0
+    .word 0
     .word z_arm_svc
-    .word z_arm_reserved
+    .word 0
 #elif defined(CONFIG_ARMV7_M_ARMV8_M_MAINLINE)
     .word z_arm_mpu_fault
     .word z_arm_bus_fault
     .word z_arm_usage_fault
 #if defined(CONFIG_ARM_SECURE_FIRMWARE)
     .word z_arm_secure_fault
 #else
+#if defined(CONFIG_ARMV8_M_SE)
     .word z_arm_reserved
+#else
+    .word 0
+#endif /* CONFIG_ARMV8_M_SE */
 #endif /* CONFIG_ARM_SECURE_FIRMWARE */
-    .word z_arm_reserved
-    .word z_arm_reserved
-    .word z_arm_reserved
+    .word 0
+    .word 0
+    .word 0
     .word z_arm_svc
     .word z_arm_debug_monitor
 #else
 #error Unknown ARM architecture
 #endif /* CONFIG_ARMV6_M_ARMV8_M_BASELINE */
-    .word z_arm_reserved
+    .word 0
     .word z_arm_pendsv
+#if defined(CONFIG_CPU_CORTEX_M_HAS_SYSTICK)
 #if defined(CONFIG_SYS_CLOCK_EXISTS)
     .word z_clock_isr
 #else
     .word z_arm_reserved
```

Comment:
I am not sure `z_arm_reserved` is the (semantically) correct name for it.

In this patch, we are setting all the "actually reserved" vectors to 0 and the valid-but-unused ones to `z_arm_reserved`, which is a little confusing.

Reply:
Agree - z_arm_reserved, probably was named like this long long ago, because it was meant to be written in the ARM Reserved entries. Feel free to suggest a different name.

Reply:
Maybe `z_arm_exc_spurious`, like exception-equivalent of `z_irq_spurious`?

Reply:
maybe yes, but let me think a bit more about it.
It should be out of the scope of this PR, which basically tries to address an issue with LPC5569 that @MaureenHelm  has raised.

Reply:
Changed to z_arm_exc_spurious, thanks @stephanosio